### DISCUSSION TITLE
How to dynamically rewrite `index.md` into `README.md`?

### DISCUSSION THREAD

**Discussion by Octobug** [vuejs/vitepress#1942] [Q&A] [answered]
From this PR #1798, which solved issue #280 , I know I could rewrite ONE `index.md` into `README.md` by:

```js
export default {
  rewrites: {
    'dir/README.md': 'dir/index.md',
  }
}
```

What if I have so many `README.md`s to rewrite? How to write a dynamic mapping?

Can anyone provide an example for this case? Thanks!

**Comment by brc-dd** [comment, marked as answer]
`':foo*/README.md': ':foo*/index.md'`

**Reply by Octobug** [reply]
It works fine, thank you!

**Comment by Octobug** [comment]
Hi, @brc-dd ,

If I have a deeper directory structure, I have to respectively write mappings for each depth as below:

```js
{
  rewrites: {
    'README.md': 'index.md',
    ':a*/README.md': ':a*/index.md',
    ':a*/:b*/README.md': ':a*/:b*/index.md',
    // deeper directories, more mappings...
  },
}
```

This is acceptable for now, but I wonder if there is a better way to do that. Could there be a configurable entry filename for this purpose?

I have also done some searching in [path-to-regexp](https://github.com/pillarjs/path-to-regexp). And have tried to modify VitePress's relevant source code in 

https://github.com/vuejs/vitepress/blob/5c9b75e325c27f63373c969e16035a9df5292cc9/src/node/config.ts#L434-L440


This worked functionally, but took an intolerable time to finish the `urlMatch(src)` if `src` was a long string (eg. `guide/migration-from-vitepress-0.md`). This poor performance is `path-to-regexp`'s problem.

```js
function rewrite(src: string, from: string, to: string) {
  // remove '/' which is a delimiter by default ('/#?'), so it could match any depth of path.
  const options = { delimiter: '#?' }
  const urlMatch = match(from, options)
  const res = urlMatch(src)
  if (!res) return false
  const toPath = compile(to, options)
  return toPath(res.params)
}
```

**Reply by brc-dd** [reply]
Ah, I thought just putting `'README.md': 'index.md', ':a*/README.md': ':a*/index.md'` would've worked for all nested directories as well. This was the behavior in `path-to-regexp@5`, but looks like they have changed this in their v6 (https://github.com/pillarjs/path-to-regexp/issues/214)

**Reply by Octobug** [reply]
Yep, this PR https://github.com/pillarjs/path-to-regexp/pull/270 was opened long ago, but it seemed that they won't fix it within a short time.

**Reply by brc-dd** [reply]
Yeah, there are issues on the WICG front. Sort of nothing seems standard with path matching. vue/react/next/netlify router each have a differently patched versions of the module. For now, IG you can go with the more mappings approach. You have to write them just once and I don't think there will be more than 5-6 entries there.

**Reply by Octobug** [reply]
Right, this is acceptable for me.
